DataDog/dd-trace-rb
# [NO-TICKET] Fix gem permissions in gitlab build-gem step

**What does this PR do?**

This PR is a follow-up to
https://github.com/DataDog/dd-trace-rb/pull/3531: In that PR we added validation for the permissions of the files in the packaged gem.

After merging that PR, @alexjf noticed that the GitLab CI `build-gem` job [was failing with](https://gitlab.ddbuild.io/DataDog/apm-reliability/dd-trace-rb/-/jobs/463401683):

```
Unexpected permissions for CHANGELOG.md inside pkg/ddtrace-1.21.0.dev.bfc613b.glci463401683.g20ea0f3a.gem (got 666, expected 644)
```

These permissions are indeed incorrect and don't match the ones we expect on actual gem releases.

Thus, I've sprinkled a `chmod` call to fix the permissions before packaging the gem in gitlab.

You can see in
https://gitlab.ddbuild.io/DataDog/apm-reliability/dd-trace-rb/-/jobs/463581213 that it now works fine:

```
gem release process (after packaging)
  sets the right permissions on the gem files
Finished in 0.07305 seconds (files took 1.2 seconds to load)
```

**Motivation:**

Fix GitLab CI pipeline.

**Additional Notes:**

I guess this check is already flagging permission issues! Great success :)

**How to test the change?**

See above -- I manually triggered the CI pipeline with this change and confirmed it passes.

**For Datadog employees:**
credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.

This PR actually does touch code that publishes internal builds, but these are not made available to users.
